sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/JdbcUtils.java

```diff
@@ -0,0 +1,119 @@
+import java.sql.Types;
+import java.util.HashMap;
+import java.util.Map;
+
+public class JdbcUtils {
+
+    private static final Map<Integer, String > RS_TYPE_MAPPING = new HashMap<>();
+
+    /**
+     *
+     */
+    static {
+        RS_TYPE_MAPPING.put(Types.BIT, "BIT");
+        RS_TYPE_MAPPING.put(Types.TINYINT, "TINYINT");
+        RS_TYPE_MAPPING.put(Types.BIGINT, "BIGINT");
+        RS_TYPE_MAPPING.put(Types.LONGVARBINARY, "LONGVARBINARY");
+        RS_TYPE_MAPPING.put(Types.VARBINARY, "VARBINARY");
+        RS_TYPE_MAPPING.put(Types.BINARY, "BINARY");
+        RS_TYPE_MAPPING.put(Types.LONGVARCHAR, "LONGVARCHAR");
+        RS_TYPE_MAPPING.put(Types.NULL, "NULL");
+        RS_TYPE_MAPPING.put(Types.CHAR, "CHAR");
+        RS_TYPE_MAPPING.put(Types.NUMERIC, "NUMERIC");
+        RS_TYPE_MAPPING.put(Types.DECIMAL, "DECIMAL");
+        RS_TYPE_MAPPING.put(Types.INTEGER, "INTEGER");
+        RS_TYPE_MAPPING.put(Types.SMALLINT, "SMALLINT");
+        RS_TYPE_MAPPING.put(Types.FLOAT, "FLOAT");
+        RS_TYPE_MAPPING.put(Types.REAL, "REAL");
+        RS_TYPE_MAPPING.put(Types.DOUBLE, "DOUBLE");
+        RS_TYPE_MAPPING.put(Types.VARCHAR, "VARCHAR");
+        RS_TYPE_MAPPING.put(Types.DATE, "DATE");
+        RS_TYPE_MAPPING.put(Types.TIME, "TIME");
+        RS_TYPE_MAPPING.put(Types.TIMESTAMP, "TIMESTAMP");
+        RS_TYPE_MAPPING.put(Types.OTHER, "OTHER");
+    }
+
+    private static Schema.FieldType fromFieldType(String typeName) { //List<TableFieldSchema> nestedFields
+        switch (typeName) {
+            case "VARCHAR":
+            case "CHAR":
+            case "LONGVARCHAR":
+            case "CLOB":
+                return Schema.FieldType.STRING;
+            case "BIT":
+                return Schema.FieldType.BOOLEAN;
+            case "NUMERIC":
+                return Schema.FieldType.INT64;
+            case "TINYINT":
+                return Schema.FieldType.BYTE;
+            case "SMALLINT":
+                return Schema.FieldType.INT16;
+            case "INTEGER":
+                return Schema.FieldType.INT32;
+            case "BIGINT":
+                return Schema.FieldType.INT64;
+            case "REAL":
+            case "FLOAT":
+                return Schema.FieldType.FLOAT;
+            case "DOUBLE":
+                return Schema.FieldType.DOUBLE;
+            case "VARBINARY":
+            case "BINARY":
+            case "BLOB":
+                return Schema.FieldType.BYTES;
+            case "DATE":
+                return Schema.FieldType.logicalType(
+                        new LogicalTypes.PassThroughLogicalType<Instant>(
+                                "SqlDateType", "", Schema.FieldType.DATETIME) {});
+            case "TIME":
+                return Schema.FieldType.logicalType(
+                        new LogicalTypes.PassThroughLogicalType<Instant>(
+                                "SqlTimeType", "", Schema.FieldType.DATETIME) {});
+            case "TIMESTAMP":
```

Comment:
FYI: these timestamps can have nanosecond precision which is apparently not supported by Beam yet. Bear that in mind when writing the value conversion logic.

Reply:
@charithe Not sure, if I understand this exactly. For Date, Time and TimeStamp data types, Shouldn't we be using the same `Schema.FieldType.logicalType` function for conversion?